velox/functions/lib/aggregates/SetAggregateBase.h

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Aggregate.h"
+#include "velox/exec/SetAccumulator.h"
+#include "velox/functions/lib/CheckNestedNulls.h"
+
+using namespace facebook::velox::aggregate::prestosql;
+
+namespace facebook::velox::functions::aggregate {
+
+template <typename T, bool ignoreNulls = false>
+class SetBaseAggregate : public exec::Aggregate {
```

Comment:
nit: could we align the class name `SetBaseAggregate` and the file name `SetAggregateBase.h`?